docs/useListController.md

```diff
@@ -85,6 +88,30 @@ const MyList = ({
 };
 ```
 
+## `storeKey`
+
+To display multiple lists of the same resource and keep distinct store states for each of them (filters, sorting and pagination), specify unique keys with the `storeKey` property.
+
+In case no `storeKey` is provided, the states will be stored with the following key: `${resource}.listParams`.
+
+**Note:** Please note that selection state will remain linked to a resource-based key as described [here](./List.md#disablesyncwithlocation).
+
+```jsx
+// display the top 5 posts
+<ListController
+    resource="posts"
+    storeKey="top"
+    sort={{ field: 'votes', order: 'DESC' }}
+    children={params => (
+        <ul>{
+            !params.isLoading && params.data.map(post => (
+                <li key={`post_${post.id}`}>{post.title} - {post.votes} votes</li>
+            ))
+        }</ul> 
+    )}
+/>
+```
```

Comment:
We should use the `useListController` here

Besides, this section should be copied to the `<List>` component documentation - that's where people will look for it.

Reply:
Done.

This was the opportunity to detect and fix a bug in `List.tsx`: `storeKey` was not passed down to the `useListController`.

I also updated the `useListController` and `List` storybooks to have stories closer to real use-cases (IMO).

Let me know what you think!

Reply:
Now that every List component can have its own key, can we turn off storing/reading pagination details using query string parameters? Making the query string parameters mandatory for each list view means that you can't functionally have more than one List on a given page and also turn on pagination, without breaking every List component on the page. Filtering has the same problem. Adding column sorting, pagination or filtering to any List component will apply the same settings to every other List on that page.

Reply:
@crates  See my comment [here](https://github.com/marmelab/react-admin/issues/8038#issuecomment-1245004316)

Reply:
@slax57 - Thanks for following up, Jean-Baptiste! That prop was exactly what I was looking for, and solved everything except the part that this upcoming PR will resolve. You guys rock!